### PR TITLE
fix: Partition system refactor

### LINKED ISSUE #1109: Jumping to a world, then back to the new SDK7 plaza does not load
### **Explorer Alpha build version:**
Main

### **Issue Description:**
Jumping to a world i.e. /goto metadynelabs and then using /goto genesis to jump back results in the SDK7 scene not loading.

### **Steps to reproduce:**

  1. /goto metadynelabs
  2. /goto genesis
  3. Notice no scene

The issue is not limited to teleporting to worlds, it can also be reproduced by teleporting to far away spots.
Other steps:
1. `/goto -149,-150`
2. After spawning, go back to Genesis Plaza with `/goto 0,0`
3. Alternatively, we can also check the issue by teleporting with the map.

https://github.com/decentraland/unity-explorer/assets/139157886/36e1d766-7f7e-4d0d-9cc3-85741ccfc469

https://github.com/decentraland/unity-explorer/assets/139157886/28da02d0-6b73-4c16-b20f-6de20790e71d

### PR DESCRIPTION
**Pull request by dalkia**
## What does this PR change?

Fix #1109 

- Introduced `PartitionDataContainer`. 
-  This allows to clear the partition array on realm change, therefore resetting and removing all possible dirty states from previous realms.
- Also, refactored the system so it doesnt do partition on every frame, but rather on the moment that the camera has changed. Adjusted the dirtyness condition to follow this logic
- Added the `[BurstCompile]` tag to the partition job

## How to test the changes?

1. Launch the explorer
2. Teleport back and forth between realms. Everything should be consistent

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by Kinerius — APPROVED** [review]
I approve the change that I suggested and agreed to do
image

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/PartitionSceneEntitiesSystem.cs [hunk 1]
```diff
@@ -140,7 +108,7 @@ private void PartitionNewEntity(in Entity entity, ref SceneDefinitionComponent d
             }
             else
             {
-                PartitionData partition = partitions[definition.InternalJobIndex];
+                var partition = partitionDataContainer.Partitions[definition.InternalJobIndex];
```
Comment (mikhail-dcl) [inline Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/PartitionSceneEntitiesSystem.cs:111]: Can we call `SetPartitionComponentData` instead? Seems to be this piece is duplicated

**Review by mikhail-dcl — APPROVED** [review]
Looks good to me but I have minor suggestions

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/PartitionSceneEntitiesSystem.cs (hunk 1 above)
Comment (dalkia) [reply Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/PartitionSceneEntitiesSystem.cs:111]: Of course, changing

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/Components/PartitionDataContainer.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        {
+            this.deployedSceneLimit = deployedSceneLimit;
+
+            // TODO: This might change with quality settings, consider updating them
+            sqrDistanceBuckets = new NativeArray<int>(partitionSettingsSqrDistanceBuckets.Count, Allocator.Persistent);
+
+            for (var i = 0; i < partitionSettingsSqrDistanceBuckets.Count; i++)
+                sqrDistanceBuckets[i] = partitionSettings.SqrDistanceBuckets[i];
+
+            Restart();
```
[the remark this answers is not in the document]
Comment (dalkia) [reply Explorer/Assets/Scripts/ECS/SceneLifeCycle/Components/PartitionDataContainer.cs:47]: Sounds about right. I added the `partitionDataContainer.Restart();` at the end of SetRealm in RealmController.
And removed entirely the `Clear()`

**Pull request merged**